crates/rune/src/runtime/function.rs

```diff
@@ -520,21 +521,17 @@ where
             }
             Inner::FnTupleStruct(tuple) => {
                 vm_try!(check_args(args.count(), tuple.args));
-                vm_try!(Value::tuple_struct(
-                    tuple.rtti.clone(),
-                    vm_try!(args.try_into_vec())
-                ))
+                let (vec, _guard) = vm_try!(unsafe { args.unsafe_into_vec() });
```

Comment:
Could you add test cases for these?

What should happen if a reference is used is that the value tuple returned contains values which return `false` when checked with [`Value::is_readable` and `Value::is_writable`](https://github.com/rune-rs/rune/blob/999eac032df1148baadde61fc412bebd23630b12/crates/rune/src/runtime/value.rs#L555).

Reply:
Just a quick note that I do intend to do this at some point. I'm just really busy with summer travel and other work, so it may be a few weeks.

Reply:
I think the tests I added are what you were asking for, but let me know if I misunderstood. Sorry for the delay 🙃

Reply:
So I'm pretty sure this never works, because the guard cannot possibly ever be held for the duration of which the tuple you constructed is used.

I don't think it's unsound though. It just means the values contained in the tuple will never be accessible. Writing a test case which uses this call function and tries to use the interior value somehow should reveal this.

Reply:
Ah, I think I see what you're saying now. In that case, should call fail for FnTuple{Struct,Variant} if there are references, or do we need some other signature that excludes them for GuardedArgs?